eventsource/stores/sqlstore/store.go

```diff
@@ -32,19 +32,21 @@ func New(db *sql.DB, tableName string) eventsource.Store {
 	}
 }
 
-func (store *store) fetchRecords(ctx context.Context, query string, limit int, args ...interface{}) (records []eventsource.Record, err error) {
+func (store *store) fetchRecords(ctx context.Context, query string, limit int, args ...interface{}) ([]eventsource.Record, error) {
 	var limitStr string
 	if limit == 0 {
 		limitStr = "ALL"
 	} else {
 		limitStr = strconv.Itoa(limit)
 	}
 
+	records := []eventsource.Record{}
 	stmt, err := store.db.PrepareContext(ctx, fmt.Sprintf(query, store.tablename, limitStr))
 	if err != nil {
 		err = errors.Wrap(err, "failed to prepare sql query")
-		return
+		return records, err
```

Comment:
maybe `return []eventsource.Record{}, err ` to be more explicit

Reply:
yeah I think it is cleaner but as you say this is the old behaviour. SO maybe we leave it in for now